fc-community-backend/apps/api/src/factchain-core/utils.ts

```diff
@@ -15,11 +15,15 @@ export function timePeriodToBlockPeriods(
   }
 
   const currentDate = new Date();
```

Comment:
new Date() will create a date object using the local timezone of the server where the code is running. If the Ethereum blocks are timestamped in UTC or another timezone it will not work as intended.

Reply:
I would use a DateTime library as MomentJS or others. Could be great to benchmark and choose one to use everywhere so time is always handled the same way. You may know but time handling is a nightmare.

Reply:
Thanks for mentioning MomentJS. It's a great library we should use!